components/hab/src/command/origin/invitations/send.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{api_client::{self,
+                         Client},
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+use reqwest::StatusCode;
+
+pub fn start(ui: &mut UI,
+             bldr_url: &str,
+             origin: &str,
+             token: &str,
+             invitation_account: &str)
+             -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Sending,
+              format!("invitation to user {} for origin {}",
+                      invitation_account, origin))?;
+
+    match api_client.send_origin_invitation(origin, token, invitation_account) {
+        Ok(_) => {
+            ui.status(Status::Sent, "the invitation successfully!".to_string())
+              .or(Ok(()))
+        }
+        Err(err @ api_client::Error::APIError(StatusCode::FORBIDDEN, _)) => {
+            ui.fatal("Failed to send the invitation!")?;
+            ui.fatal("This situation could arise, if for example, you are not a member of the \
+                      origin.")?;
+            Err(Error::APIClient(err))
+        }
+        Err(err @ api_client::Error::APIError(StatusCode::UNAUTHORIZED, _)) => {
+            ui.fatal("Failed to send the invitation!")?;
+            Err(Error::APIClient(err))
+        }
+        Err(err @ api_client::Error::APIError(StatusCode::NOT_FOUND, _)) => {
+            ui.fatal("Failed to send the invitation!")?;
+            ui.fatal("The origin or the member being invitationd (or both) does not exist.")?;
```

Comment:
"invitationd" -> "invitation"

The same "or both" observation applies here, as well.

Reply:
Thanks for spotting this! I corrected in https://github.com/habitat-sh/habitat/pull/7349/commits/cb0ca5bef10de78cad9190a7a17991a21cb5b7d7